### PR TITLE
fix: ERROR when agent finishes on final iteration (#2659)

### PR DESCRIPTION
**Pull request by trayanmomkov**
## Summary

  - If the agent set FINISHED during its final allowed step, preserve that status instead of overwriting with ERROR                                                                                                         
  - Add regression test for the boundary case   

## Checklist

- [x] Tests — Yes, added test_execution_status_finished_on_final_iteration regression test, all 11 tests pass.                 
- [ ] Example — N/A, no example involved.
- [ ] Instructions — N/A, no new instructions to follow.                                                                       
- [ ] Documentation — N/A, this is a small bug fix, no docs needed.

### CODE REVIEW

File: tests/sdk/conversation/local/test_agent_status_transition.py [hunk 1]
```diff
@@ -475,3 +475,72 @@ def _make_tool(conv_state=None, **params) -> Sequence[ToolDefinition]:
...
+
+
+def test_execution_status_finished_on_final_iteration():
+    """FINISHED is preserved when agent completes on its final iteration.
+
+    Regression test for: agent's FINISHED status being overwritten with
+    ERROR when the task completes exactly on the max_iteration_per_run
+    boundary.
+    """
+    from openhands.sdk.event.conversation_error import ConversationErrorEvent
```
Comment (VascoSch92) [inline tests/sdk/conversation/local/test_agent_status_transition.py]: nit: import on top of the file

File: tests/sdk/conversation/local/test_agent_status_transition.py [hunk 2]
```diff
@@ -475,3 +475,72 @@ def _make_tool(conv_state=None, **params) -> Sequence[ToolDefinition]:
...
+                arguments='{"command": "test_command"}',
+                origin="completion",
+            )
+        ],
+    )
+    finish_message = Message(
+        role="assistant", content=[TextContent(text="Task completed successfully")]
+    )
+
+    llm = TestLLM.from_messages(
```
Comment (VascoSch92) [inline tests/sdk/conversation/local/test_agent_status_transition.py:513]: nice

**Review by VascoSch92 — APPROVED** [review]
just a small nit but otherwise LGTM
